mm2src/coins/solana/solana_coin.rs

```diff
@@ -169,11 +193,11 @@ impl SolanaCoin {
     pub(crate) async fn rpc_client(&self) -> MmResult<Arc<RpcClient>, String> {
         let mut rpcs = self.rpc_clients.lock().await;
 
-        if let Some(index) = rpcs.iter().position(|rpc| rpc.get_health().is_ok()) {
-            // Put healthy one to the front.
-            rpcs.rotate_left(index);
-
-            return Ok(rpcs[0].clone());
+        for (index, rpc) in rpcs.iter().enumerate() {
+            if rpc.get_health().await.is_ok() {
+                rpcs.rotate_left(index);
+                return Ok(rpcs[0].clone());
+            }
         }
```

Comment:
A comment for next PRs, please think about randomizing RPCs order on coin activation.

Reply:
> To not let all users use the first one and overload it, load should be distributed. At least it was done like that for EVMs.

This would kill the ability of controlling the prioritization of the nodes we add to the application (e.g., having the paid node as the last backup and the most stable one as the primary/main node).

Reply:
I understand this, we should also have a way to prioritize them into primary and secondary, there was an issue for this for other coins https://github.com/KomodoPlatform/komodo-defi-framework/issues/1520
But all this is very low priority, so keep it as is for now.